src/main/scala/mesosphere/marathon/state/PortDefinition.scala

```diff
@@ -14,7 +14,7 @@ case class PortDefinition(
 
 object PortDefinition {
   implicit val portDefinitionValidator = validator[PortDefinition] { portDefinition =>
-    portDefinition.protocol is oneOf(DiscoveryInfo.Port.AllowedProtocols)
+    portDefinition.protocol.split(',').toIterable is every(oneOf(DiscoveryInfo.Port.AllowedProtocols))
```

Comment:
It's not clear to me why we restrict this field at all, if it's only used
for DI

On Thu, Sep 1, 2016 at 12:07 PM, Matthias Veit <notifications@github.com>
wrote:

> In src/main/scala/mesosphere/marathon/state/PortDefinition.scala
> <https://github.com/mesosphere/marathon/pull/4284#discussion_r77204102>:
>
> > @@ -14,7 +14,7 @@ case class PortDefinition(
> >
> >  object PortDefinition {
> >    implicit val portDefinitionValidator = validator[PortDefinition] { portDefinition =>
> > -    portDefinition.protocol is oneOf(DiscoveryInfo.Port.AllowedProtocols)
> > +    portDefinition.protocol.split(',').toIterable is every(oneOf(DiscoveryInfo.Port.AllowedProtocols))
>
> @jdef <https://github.com/jdef>: Is there a term which describes
> tcp_and_udp? So we could use this as a separate enum value.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/mesosphere/marathon/pull/4284/files/a1c71b58309775a3ca78fc02af2fe987bc5ee603#r77204102>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ACPVLNF6WZnUwynQcs8nhR1BYHGX3QVFks5qlvgugaJpZM4Jysrn>
> .
>